hack/boilerplate/boilerplate.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var (
+	skippedPaths   = regexp.MustCompile(`Godeps|third_party|_gopath|_output|\.git|cluster/env.sh|vendor|test/e2e/generated/bindata.go|site/themes/docsy`)
+	boilerplatedir = flag.String("boilerplate-dir", ".", "Boilerplate directory for boilerplate files")
+	rootdir        = flag.String("rootdir", "../../", "Root directory to examine")
+	verbose        = flag.Bool("v", false, "Verbose")
+)
+
+func main() {
+	flag.Parse()
+	refs, err := extensionToBoilerplate(*boilerplatedir)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if len(refs) == 0 {
+		log.Fatal("no references in ", *boilerplatedir)
+	}
+	files, err := filesToCheck(*rootdir, refs)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		pass, err := filePasses(file, refs[filepath.Ext(file)])
+		if err != nil {
+			log.Println(err)
+		}
+		if !pass {
+			path, err := filepath.Abs(file)
+			if err != nil {
+				log.Println(err)
+			}
+			fmt.Println(path)
+		}
+	}
+
+}
+
+// extensionToBoilerplate returns a map of file extension to required boilerplate text.
+func extensionToBoilerplate(dir string) (map[string][]byte, error) {
+	refs := make(map[string][]byte)
+	files, _ := filepath.Glob(dir + "/*.txt")
+	for _, filename := range files {
+		re := regexp.MustCompile(`\.txt`)
```

Comment:
Rather than compiling this within a for loop, it should be made a top-level `var`, like you did with `skippedPaths`